app/views/core/RootView.coffee

```diff
@@ -104,9 +104,16 @@ module.exports = class RootView extends CocoView
   switchToStudentMode: ->
     text = 'Switching to test student account..'
     noty({ text, type: 'success', timeout: 5000, killer: true })
-    me.switchToStudentMode()
-      .then(() -> window.location.reload())
-      .catch((err) -> errors.showNotyNetworkError(err))
+    me.getTestStudentId()
+      .then (student) =>
+        if student.new
+          @openNewTestStudentModal(student.id)
```

Comment:
let's open this modal thrice instead of only when it's new so that teachers without class can come back and join the class easily. Also, for existing users, a good idea to show them. Let's save the value in localStorage